src/core/types/object/index.js

```diff
@@ -210,6 +212,7 @@ export function getValueForExpr(obj, expr) {
   }
   // Otherwise, navigate via properties.
   const parts = expr.split('.');
+  /** @type {*} */
```

Comment:
nit: this cast would look better surrounding obj right? or does it need to be on the binding?

Reply:
This is declaring that `value` can be any type, otherwise assigning `obj` (which is a `JsonObject`) tells TS to assume `value` will always be a `JsonObject`, which will fail lower down when doing `value = value[part]`. We could surround `obj`, though my inclination is toward declaring the type of the variable directly when ambiguous, rather than telling TS to infer that the type is `any` based on casting the assigned value. It's an incredibly minor difference with no impact, though.